src/index.js

```diff
@@ -61,43 +30,30 @@ async function setupConnection () {
     updateCfg = true
   }
 
-  const ipfsd = await createDaemon(config)
-
-  // createDaemon has changed the config object,
-  // but didn't add the repo variable.
-  if (updateCfg) {
-    config.path = ipfsd.repoPath
-    store.set('config', config)
-  }
-
-  return ipfsd
-}
-
-async function run () {
-  try {
-    await app.whenReady()
-  } catch (e) {
-    dialog.showErrorBox('Electron could not start', e.stack)
-    app.exit(1)
-  }
-
   try {
     // Initial context object
     let ctx = {
-      ipfsd: await setupConnection()
+      ipfsd: await createDaemon(config)
+    }
+
+    // createDaemon has changed the config object,
+    // but didn't add the repo variable.
+    if (updateCfg) {
```

Comment:
Let's simplify this. `createDaemon` shouldn't mutate the config that is passed to it. Let's pull all the setting of defaults up into this file, so we don't have to guess about when we might need to update the stored config.

Reply:
@olizilla then we have to test if there is an IPFS_PATH env variable and if not, use the home path. All of that is already done by ipfsd-ctl which will get updated in case of something changes. I just didn't want to repeat the code.

Reply:
That's totally reasonable. How would you re-arrange things so that createDaemon doesn't mutate the options object that is passed to it? How can we make the code clearer that we may provide a path, but if we dont, we want to ask the ipfsd what it's path is?

Reply:
Simple: we could initially put the config settings as sth like:

```json
{
	"type": "go",
	"path": "",
	"flags": [...defaultFlags]
}
```

`createDaemon` would not replace the settings with the defaults and after getting the `ipfsd` object we could do:

```javascript
if (cfg.path === '') {
	cfg.path = ipfsd.repoPath
	// update cfg file...
}
```

Reply:
@olizilla just updated the code with a clearer version and this way `createDaemon` doesn't change the opts object.